app/assets/javascripts/controllers/schedules/controller.js

```diff
@@ -6,6 +6,40 @@
  * @memberOf Yacs.views
  */
 Yacs.views.schedules = function (target, params) {
+  // before doing anything, determine how to use the route
+  // parameters to choose the section ids to be passed to the
+  // API
+  var schedule_ids = [];
+
+  // check for query parameters
+  if('section_ids' in params) {
+    // if there are query parameters,
+    // use them and ignore current selections
+    var section_ids = params['section_ids'].split(',');
+    // eliminate all non-integer section ids
+    var sidlen = section_ids.length;
```

Comment:
I can't think of any reason these need to be number types, it should be fine to leave them as string throughout their lifecycle.